diff -Nudr -U3 -x .svn libmodplug.pristine/src/fastmix.cpp libmodplug/src/fastmix.cpp
--- libmodplug.pristine/src/fastmix.cpp	2009-12-31 16:32:40.000000000 +0000
+++ libmodplug/src/fastmix.cpp	2009-12-31 16:55:18.000000000 +0000
@@ -1881,13 +1881,13 @@
 			vumax = n;
 		p = n >> (8-MIXING_ATTENUATION) ; // 24-bit signed
 #if defined(WORDS_BIGENDIAN)
-		buf[i*3+0] = p & 0xFF0000 >> 24;
-		buf[i*3+1] = p & 0x00FF00 >> 16 ;
-		buf[i*3+2] = p & 0x0000FF ;
+		buf[i*3+0] = (p >> 16) & 0xFF;
+		buf[i*3+1] = (p >> 8)  & 0xFF;
+		buf[i*3+2] = (p >> 0)  & 0xFF;
 #else
-		buf[i*3+0] = p & 0x0000FF ;
-		buf[i*3+1] = p & 0x00FF00 >> 16;
-		buf[i*3+2] = p & 0xFF0000 >> 24;
+		buf[i*3+0] = (p >> 0)  & 0xFF;
+		buf[i*3+1] = (p >> 8)  & 0xFF;
+		buf[i*3+2] = (p >> 16) & 0xFF;
 #endif
 	}
 	*lpMin = vumin;
