diff -Nudr -U3 libmodplug.really-pristine/src/libmodplug/stdafx.h libmodplug.pristine/src/libmodplug/stdafx.h
--- libmodplug.really-pristine/src/libmodplug/stdafx.h	2009-04-20 08:09:15.000000000 +0100
+++ libmodplug.pristine/src/libmodplug/stdafx.h	2009-12-31 15:45:20.000000000 +0000
@@ -51,6 +51,29 @@
 #include <malloc.h>
 #endif
 
+#ifdef __amigaos__
+
+/* On AmigaOS platforms we work around the fact that the toolchain headers
+ * pollute the namespace with many MSVC-like type declarations. Using their
+ * versions directly is not possible, as some are just plain wrong. WORD, for
+ * example, is implicitly signed (not unsigned) as with the BYTE type. All
+ * other toolchains do this differently, which ModPlug depends on internally.
+ */
+#define ULONG _ULONG
+#define LONG _LONG
+#define WORD _WORD
+#define BYTE _BYTE
+#define BOOL _BOOL
+
+#else // !__amigaos__
+
+/* On non-AmigaOS platforms we also define VOID; on AmigaOS we can
+ * trust the VOID declaration and the same _VOID trick doesn't work.
+ */
+typedef void VOID;
+
+#endif // __amigaos__
+
 typedef int8_t CHAR;
 typedef uint8_t UCHAR;
 typedef uint8_t* PUCHAR;
@@ -71,7 +94,6 @@
 typedef uint16_t* LPWORD;
 typedef const char* LPCSTR;
 typedef void* PVOID;
-typedef void VOID;
 
 inline LONG MulDiv (long a, long b, long c)
 {
