diff -Nudr -U3 mzx281d/contrib/libmodplug/src/sndfile.cpp mzx281d.patched/contrib/libmodplug/src/sndfile.cpp
--- mzx281d/contrib/libmodplug/src/sndfile.cpp	Sun Dec 10 19:17:07 2006
+++ mzx281d.patched/contrib/libmodplug/src/sndfile.cpp	Mon Dec 11 11:00:42 2006
@@ -6,8 +6,8 @@
 */
 
 #include <math.h>       //for GCCFIX
-#include <libmodplug/stdafx.h>
-#include <libmodplug/sndfile.h>
+#include "libmodplug/stdafx.h"
+#include "libmodplug/sndfile.h"
 
 #define MMCMP_SUPPORT
 
diff -Nudr -U3 mzx281d/contrib/libmodplug/src/sndmix.cpp mzx281d.patched/contrib/libmodplug/src/sndmix.cpp
--- mzx281d/contrib/libmodplug/src/sndmix.cpp	Mon May 23 09:50:08 2005
+++ mzx281d.patched/contrib/libmodplug/src/sndmix.cpp	Mon Dec 11 11:01:19 2006
@@ -4,8 +4,8 @@
  * Authors: Olivier Lapicque <olivierl@jps.net>
 */
 
-#include <libmodplug/stdafx.h>
-#include <libmodplug/sndfile.h>
+#include "libmodplug/stdafx.h"
+#include "libmodplug/sndfile.h"
 
 #ifdef MODPLUG_TRACKER
 #define ENABLE_STEREOVU
diff -Nudr -U3 mzx281d/contrib/libmodplug/src/tables.cpp mzx281d.patched/contrib/libmodplug/src/tables.cpp
--- mzx281d/contrib/libmodplug/src/tables.cpp	Mon May 23 09:50:08 2005
+++ mzx281d.patched/contrib/libmodplug/src/tables.cpp	Mon Dec 11 11:01:40 2006
@@ -4,8 +4,8 @@
  * Authors: Olivier Lapicque <olivierl@jps.net>
  */
 
-#include <libmodplug/stdafx.h>
-#include <libmodplug/sndfile.h>
+#include "libmodplug/stdafx.h"
+#include "libmodplug/sndfile.h"
 
 #ifndef MODPLUG_FASTSOUNDLIB
 //#pragma data_seg(".tables")
