diff -Nudr -U3 libmodplug-0.7/src/libmodplug/sndfile.h libmodplug/src/libmodplug/sndfile.h
--- libmodplug-0.7/src/libmodplug/sndfile.h	2005-07-06 18:59:32.842085856 +0100
+++ libmodplug/src/libmodplug/sndfile.h	2005-07-06 18:48:45.026568784 +0100
@@ -22,7 +22,7 @@
 
 #define MOD_AMIGAC2			0x1AB
 #define MAX_SAMPLE_LENGTH	16000000
-#define MAX_SAMPLE_RATE		50000
+#define MAX_SAMPLE_RATE		192000
 #define MAX_ORDERS			256
 #define MAX_PATTERNS		240
 #define MAX_SAMPLES			240
