diff -Nudr -x .svn libmodplug.pristine/src/fastmix.cpp libmodplug/src/fastmix.cpp
--- libmodplug.pristine/src/fastmix.cpp	2009-12-31 16:28:57.000000000 +0000
+++ libmodplug/src/fastmix.cpp	2009-12-31 16:20:25.000000000 +0000
@@ -1880,7 +1880,7 @@
 		else if (n > vumax)
 			vumax = n;
 		p = n >> (8-MIXING_ATTENUATION) ; // 24-bit signed
-#ifdef WORDS_BIGENDIAN
+#if defined(WORDS_BIGENDIAN)
 		buf[i*3+0] = p & 0xFF0000 >> 24;
 		buf[i*3+1] = p & 0x00FF00 >> 16 ;
 		buf[i*3+2] = p & 0x0000FF ;
diff -Nudr -x .svn libmodplug.pristine/src/libmodplug/sndfile.h libmodplug/src/libmodplug/sndfile.h
--- libmodplug.pristine/src/libmodplug/sndfile.h	2009-12-31 16:02:49.000000000 +0000
+++ libmodplug/src/libmodplug/sndfile.h	2009-12-31 16:18:59.000000000 +0000
@@ -517,7 +517,7 @@
 	MIDIOUT_VOLUME,
 	MIDIOUT_PAN,
 	MIDIOUT_BANKSEL,
-	MIDIOUT_PROGRAM,
+	MIDIOUT_PROGRAM
 };
 
 
@@ -1001,7 +1001,7 @@
 #define bswapBE32(X) bswap_32(ARM_get32(&X))
 
 // From libsdl
-#elif WORDS_BIGENDIAN
+#elif defined(WORDS_BIGENDIAN)
 #define bswapLE16(X) bswap_16(X)
 #define bswapLE32(X) bswap_32(X)
 #define bswapBE16(X) (X)
