Index: libmodplug/src/libmodplug/stdafx.h
===================================================================
--- libmodplug/src/libmodplug/stdafx.h	(revision 1571)
+++ libmodplug/src/libmodplug/stdafx.h	(working copy)
@@ -124,7 +124,10 @@
 
 #define GlobalFreePtr(p) free((void *)(p))
 
+#ifndef strnicmp
 #define strnicmp(a,b,c)		strncasecmp(a,b,c)
+#endif
+
 #define wsprintfA		sprintf
 
 #ifndef FALSE
