diff -Nudr -U3 libmodplug.really-pristine/src/sndmix.cpp libmodplug.pristine/src/sndmix.cpp
--- libmodplug.really-pristine/src/sndmix.cpp	2009-04-20 06:00:54.000000000 +0100
+++ libmodplug.pristine/src/sndmix.cpp	2009-12-31 15:47:12.000000000 +0000
@@ -365,7 +365,7 @@
 				// End of song ?
 				if ((m_nPattern == 0xFF) || (m_nCurrentPattern >= MAX_ORDERS))
 				{
-					//if (!m_nRepeatCount)
+					if (!m_nRepeatCount)
 						return FALSE;     //never repeat entire song
 					if (!m_nRestartPos)
 					{
@@ -398,7 +398,7 @@
 							}
 						}
 					}
-//					if (m_nRepeatCount > 0) m_nRepeatCount--;
+					if (m_nRepeatCount > 0) m_nRepeatCount--;
 					m_nCurrentPattern = m_nRestartPos;
 					m_nRow = 0;
 					if ((Order[m_nCurrentPattern] >= MAX_PATTERNS) || (!Patterns[Order[m_nCurrentPattern]])) return FALSE;
