Index: src/video/windx5/SDL_dx5events.c
===================================================================
--- src/video/windx5/SDL_dx5events.c	(revision 4067)
+++ src/video/windx5/SDL_dx5events.c	(working copy)
@@ -516,10 +516,7 @@
 
 		/* The keyboard is handled via DirectInput */
 		case WM_SYSKEYUP:
-		case WM_SYSKEYDOWN: {
-			/* Pass syskey to DefWindwoProc (ALT-F4, etc.) */
-		}
-		break;
+		case WM_SYSKEYDOWN:
 		case WM_KEYUP:
 		case WM_KEYDOWN: {
 			/* Ignore windows keyboard messages */;
