diff --git a/src/common.h b/src/common.h
index 65a9f57f..35d60e95 100644
--- a/src/common.h
+++ b/src/common.h
@@ -146,11 +146,24 @@ void __inline CLIB_DECL D_(const char *text, ...) { do {} while (0); }
 #define unlink _unlink
 #endif
 #if defined(_WIN32) || defined(__WATCOMC__) /* in win32.c */
+#define USE_LIBXMP_SNPRINTF
+/* MSVC 2015+ has C99 compliant snprintf and vsnprintf implementations.
+ * If __USE_MINGW_ANSI_STDIO is defined for MinGW (which it is by default),
+ * compliant implementations will be used instead of the broken MSVCRT
+ * functions. Additionally, GCC may optimize some calls to those functions. */
+#if defined(_MSC_VER) && _MSC_VER >= 1900
+#undef USE_LIBXMP_SNPRINTF
+#endif
+#if defined(__MINGW32__) && defined(__USE_MINGW_ANSI_STDIO) && (__USE_MINGW_ANSI_STDIO != 0)
+#undef USE_LIBXMP_SNPRINTF
+#endif
+#ifdef USE_LIBXMP_SNPRINTF
 int libxmp_vsnprintf(char *, size_t, const char *, va_list);
 int libxmp_snprintf (char *, size_t, const char *, ...);
 #define snprintf  libxmp_snprintf
 #define vsnprintf libxmp_vsnprintf
 #endif
+#endif
 
 /* Quirks */
 #define QUIRK_S3MLOOP	(1 << 0)	/* S3M loop mode */
diff --git a/src/win32.c b/src/win32.c
index 3844565a..9e900d07 100644
--- a/src/win32.c
+++ b/src/win32.c
@@ -1,10 +1,11 @@
 /* _[v]snprintf() from msvcrt.dll might not nul terminate */
 /* OpenWatcom-provided versions seem to behave the same... */
 
-#if defined(_WIN32) || defined(__WATCOMC__)
-
 #include <stdarg.h>
 #include <stdio.h>
+#include "common.h"
+
+#if defined(USE_LIBXMP_SNPRINTF)
 
 #undef snprintf
 #undef vsnprintf
