diff --git a/src/common.h b/src/common.h
index 04754cf6..01324288 100644
--- a/src/common.h
+++ b/src/common.h
@@ -120,10 +120,6 @@ typedef int tst_uint64[2 * (8 == sizeof(uint64)) - 1];
 #pragma warning(disable:4100) /* unreferenced formal parameter */
 #endif
 
-#ifndef LIBXMP_CORE_PLAYER
-#define LIBXMP_PAULA_SIMULATOR
-#endif
-
 /* Constants */
 #define PAL_RATE	250.0		/* 1 / (50Hz * 80us)		  */
 #define NTSC_RATE	208.0		/* 1 / (60Hz * 80us)		  */
diff --git a/src/format.c b/src/format.c
index 593ace13..aecde7b5 100644
--- a/src/format.c
+++ b/src/format.c
@@ -28,28 +28,14 @@
 const struct format_loader *const format_loaders[NUM_FORMATS + 2] = {
 	&libxmp_loader_xm,
 	&libxmp_loader_mod,
-#ifndef LIBXMP_CORE_DISABLE_IT
 	&libxmp_loader_it,
-#endif
 	&libxmp_loader_s3m,
-#ifndef LIBXMP_CORE_PLAYER
 	&libxmp_loader_flt,
 	&libxmp_loader_st,
 	&libxmp_loader_stm,
-	&libxmp_loader_stx,
 	&libxmp_loader_mtm,
 	&libxmp_loader_ice,
-	&libxmp_loader_imf,
-	&libxmp_loader_ptm,
-	&libxmp_loader_mdl,
 	&libxmp_loader_ult,
-	&libxmp_loader_liq,
-	&libxmp_loader_no,
-	&libxmp_loader_masi,
-	&libxmp_loader_masi16,
-	&libxmp_loader_muse,
-	&libxmp_loader_gal5,
-	&libxmp_loader_gal4,
 	&libxmp_loader_amf,
 	&libxmp_loader_asylum,
 	&libxmp_loader_gdm,
@@ -58,44 +44,10 @@ const struct format_loader *const format_loaders[NUM_FORMATS + 2] = {
 	&libxmp_loader_med2,
 	&libxmp_loader_med3,
 	&libxmp_loader_med4,
-	/* &libxmp_loader_dmf, */
-	&libxmp_loader_chip,
-	&libxmp_loader_rtm,
-	&libxmp_loader_pt3,
-	/* &libxmp_loader_tcb, */
-	&libxmp_loader_dt,
-	/* &libxmp_loader_gtk, */
-	/* &libxmp_loader_dtt, */
-	&libxmp_loader_mgt,
-	&libxmp_loader_arch,
-	&libxmp_loader_sym,
-	&libxmp_loader_digi,
-	&libxmp_loader_dbm,
-	&libxmp_loader_emod,
 	&libxmp_loader_okt,
-	&libxmp_loader_sfx,
 	&libxmp_loader_far,
-	&libxmp_loader_umx,
 	&libxmp_loader_hmn,
-	&libxmp_loader_stim,
-	&libxmp_loader_coco,
-	/* &libxmp_loader_mtp, */
-	&libxmp_loader_ims,
 	&libxmp_loader_669,
-	&libxmp_loader_fnk,
-	/* &libxmp_loader_amd, */
-	/* &libxmp_loader_rad, */
-	/* &libxmp_loader_hsc, */
-	&libxmp_loader_mfp,
-	&libxmp_loader_abk,
-	/* &libxmp_loader_alm, */
-	/* &libxmp_loader_polly, */
-	/* &libxmp_loader_stc, */
-	&libxmp_loader_xmf,
-#ifndef LIBXMP_NO_PROWIZARD
-	&libxmp_loader_pw,
-#endif
-#endif /* LIBXMP_CORE_PLAYER */
 	NULL /* list teminator */
 };
 
diff --git a/src/loaders/xm_load.c b/src/loaders/xm_load.c
index 72189600..bcc677ca 100644
--- a/src/loaders/xm_load.c
+++ b/src/loaders/xm_load.c
@@ -35,7 +35,7 @@
 
 #include "loader.h"
 #include "xm.h"
-#ifndef LIBXMP_CORE_PLAYER
+#if 0
 #include "vorbis.h"
 #endif
 
@@ -373,7 +373,7 @@ err:
  * for more details. */
 #define XM_MAX_SAMPLES_PER_INST 32
 
-#ifndef LIBXMP_CORE_PLAYER
+#if 0
 #define MAGIC_OGGS	0x4f676753
 
 static int is_ogg_sample(HIO_HANDLE *f, struct xmp_sample *xxs)
@@ -721,7 +721,7 @@ static int load_instruments(struct module_data *m, int version, HIO_HANDLE *f)
 			if (version > 0x0103) {
 			        D_(D_INFO "  read sample: index:%d sample id:%d", j, sub->sid);
 
-#ifndef LIBXMP_CORE_PLAYER
+#if 0
 				if (is_ogg_sample(f, xxs)) {
 					if (oggdec(m, f, xxs, xsh[j].length) < 0) {
 						return -1;
