diff --git a/include/xmp.h b/include/xmp.h
index 8ec6deeb..abf49cf4 100644
--- a/include/xmp.h
+++ b/include/xmp.h
@@ -1,6 +1,22 @@
 #ifndef XMP_H
 #define XMP_H
 
+/* Suppress unwanted debug messages */
+#ifdef DEBUG
+#undef DEBUG
+#define MEGAZEUX_DEBUG
+#endif
+
+/* Force libxmp to build static */
+#ifndef BUILDING_STATIC
+#define BUILDING_STATIC
+#endif
+
+/* Force libxmp to not use versioned symbols on Linux */
+#ifdef XMP_SYM_VISIBILITY
+#undef XMP_SYM_VISIBILITY
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -26,6 +42,7 @@ extern "C" {
 #elif defined(__SUNPRO_C) && defined(XMP_LDSCOPE_GLOBAL)
 # define LIBXMP_EXPORT __global
 #elif defined(EMSCRIPTEN)
+# include <emscripten.h>
 # define LIBXMP_EXPORT EMSCRIPTEN_KEEPALIVE
 #else
 # define LIBXMP_EXPORT
