diff --git a/src/common.h b/src/common.h
index 1646441..4aced6f 100644
--- a/src/common.h
+++ b/src/common.h
@@ -11,7 +11,8 @@
 
 #if defined(__GNUC__) || defined(__clang__)
 #if !defined(WIN32) && !defined(__ANDROID__) && !defined(__APPLE__) && !defined(__AMIGA__) && !defined(B_BEOS_VERSION) && !defined(__ATHEOS__) && !defined(EMSCRIPTEN) && !defined(__MINT__) 
-#define USE_VERSIONED_SYMBOLS
+/* Force libxmp to link on unix platforms */
+//#define USE_VERSIONED_SYMBOLS
 #endif
 #endif
 
