diff --git a/src/common.h b/src/common.h
index 4aced6f1..67b257cf 100644
--- a/src/common.h
+++ b/src/common.h
@@ -125,8 +125,10 @@ void __inline CLIB_DECL D_(const char *text, ...) { do {} while (0); }
 #ifdef _MSC_VER
 #define dup _dup
 #define fileno _fileno
+#if _MSC_VER <= 1800 /* VS2015+ */
 #define snprintf _snprintf
 #define vsnprintf _vsnprintf
+#endif
 #define strnicmp _strnicmp
 #define strdup _strdup
 #define fdopen _fdopen
