diff --git a/src/load.c b/src/load.c
index 401ba861..61352e5e 100644
--- a/src/load.c
+++ b/src/load.c
@@ -578,11 +578,15 @@ int xmp_load_module_from_file(xmp_context opaque, void *file, long size)
 	struct context_data *ctx = (struct context_data *)opaque;
 	struct module_data *m = &ctx->m;
 	HIO_HANDLE *h;
-	FILE *f = fdopen(fileno((FILE *)file), "rb");
+	FILE *f = (FILE *)file;
 	int ret;
 
-	if ((h = hio_open_file(f)) == NULL)
+	if ((h = hio_open_file(f)) == NULL) {
+		/* Close the provided file since hio_close will close it
+		 * in the general case. */
+		fclose(f);
 		return -XMP_ERROR_SYSTEM;
+	}
 
 	if (ctx->state > XMP_STATE_UNLOADED)
 		xmp_release_module(opaque);
