diff -Nrup _src/libmodplug/stdafx.h src/libmodplug/stdafx.h
--- _src/libmodplug/stdafx.h	2017-05-25 07:26:43.565056316 +0200
+++ src/libmodplug/stdafx.h	2017-05-25 07:27:09.834057068 +0200
@@ -124,18 +124,6 @@ static inline void ProcessPlugins(int n)
 
 #endif /* _WIN32 */
 
-#if defined(_WIN32) || defined(__CYGWIN__)
-# if defined(MODPLUG_BUILD) && defined(DLL_EXPORT)	/* building libmodplug as a dll for windows */
-#   define MODPLUG_EXPORT __declspec(dllexport)
-# elif defined(MODPLUG_BUILD) || defined(MODPLUG_STATIC)	/* building or using static libmodplug for windows */
-#   define MODPLUG_EXPORT
-# else
-#   define MODPLUG_EXPORT __declspec(dllimport)			/* using libmodplug dll for windows */
-# endif
-#elif defined(MODPLUG_BUILD) && defined(SYM_VISIBILITY)
-#   define MODPLUG_EXPORT __attribute__((visibility("default")))
-#else
 #define MODPLUG_EXPORT
-#endif
 
 #endif
diff -Nrup _src/modplug.h src/modplug.h
--- _src/modplug.h	2017-05-25 07:26:43.574056317 +0200
+++ src/modplug.h	2017-05-25 07:27:03.787056895 +0200
@@ -11,19 +11,7 @@
 extern "C" {
 #endif
 
-#if defined(_WIN32) || defined(__CYGWIN__)
-# if defined(MODPLUG_BUILD) && defined(DLL_EXPORT)	/* building libmodplug as a dll for windows */
-#   define MODPLUG_EXPORT __declspec(dllexport)
-# elif defined(MODPLUG_BUILD) || defined(MODPLUG_STATIC)	/* building or using static libmodplug for windows */
-#   define MODPLUG_EXPORT
-# else
-#   define MODPLUG_EXPORT __declspec(dllimport)			/* using libmodplug dll for windows */
-# endif
-#elif defined(MODPLUG_BUILD) && defined(SYM_VISIBILITY)
-#   define MODPLUG_EXPORT __attribute__((visibility("default")))
-#else
 #define MODPLUG_EXPORT
-#endif
 
 struct _ModPlugFile;
 typedef struct _ModPlugFile ModPlugFile;
