diff --git a/src/load_669.cpp b/src/load_669.cpp
index c275615..1920878 100644
--- a/src/load_669.cpp
+++ b/src/load_669.cpp
@@ -71,7 +71,7 @@ BOOL CSoundFile::Read669(const BYTE *lpStream, DWORD dwMemLength)
 	m_dwSongFlags |= SONG_LINEARSLIDES;
 	m_nMinPeriod = 28 << 2;
 	m_nMaxPeriod = 1712 << 3;
-	m_nDefaultTempo = 125;
+	m_nDefaultTempo = 78;
 	m_nDefaultSpeed = 6;
 	m_nChannels = 8;
 	memcpy(m_szNames[0], pfh->songmessage, 16);
@@ -154,7 +154,7 @@ BOOL CSoundFile::Read669(const BYTE *lpStream, DWORD dwMemLength)
 					case 0x02:	command = CMD_TONEPORTAMENTO; break;
 					case 0x03:	command = CMD_MODCMDEX; param |= 0x50; break;
 					case 0x04:	command = CMD_VIBRATO; param |= 0x40; break;
-					case 0x05:	if (param) command = CMD_SPEED; else command = 0; param += 2; break;
+					case 0x05:	if (param) command = CMD_SPEED; else command = 0; break;
 					case 0x06:	if (param == 0) { command = CMD_PANNINGSLIDE; param = 0xFE; }
 								else if (param == 1) { command = CMD_PANNINGSLIDE; param = 0xEF; }
 								else command = 0;
@@ -174,7 +174,7 @@ BOOL CSoundFile::Read669(const BYTE *lpStream, DWORD dwMemLength)
 				for (UINT i=0; i<8; i++) if (!mspeed[i].command)
 				{
 					mspeed[i].command = CMD_SPEED;
-					mspeed[i].param = pfh->tempolist[npat] + 2;
+					mspeed[i].param = pfh->tempolist[npat];
 					break;
 				}
 			}
