diff --git a/src/libmodplug/sndfile.h b/src/libmodplug/sndfile.h
index 091e25fd..2a3c3147 100644
--- a/src/libmodplug/sndfile.h
+++ b/src/libmodplug/sndfile.h
@@ -71,6 +71,7 @@ typedef const BYTE * LPCBYTE;
 #define MOD_TYPE_J2B		0x800000
 #define MOD_TYPE_ABC		0x1000000
 #define MOD_TYPE_PAT		0x2000000
+#define MOD_TYPE_GDM		0x40000000 // Fake type
 #define MOD_TYPE_UMX		0x80000000 // Fake type
 #define MAX_MODTYPE		24
 
@@ -652,6 +653,7 @@ public:
 	BOOL ReadMT2(LPCBYTE lpStream, DWORD dwMemLength);
 	BOOL ReadPSM(LPCBYTE lpStream, DWORD dwMemLength);
 	BOOL ReadJ2B(LPCBYTE lpStream, DWORD dwMemLength);
+	BOOL ReadGDM(LPCBYTE lpStream, DWORD dwMemLength);
 	BOOL ReadUMX(LPCBYTE lpStream, DWORD dwMemLength);
 	BOOL ReadABC(LPCBYTE lpStream, DWORD dwMemLength);
 	BOOL TestABC(LPCBYTE lpStream, DWORD dwMemLength);
diff --git a/src/sndfile.cpp b/src/sndfile.cpp
index cc84dee6..d1730910 100644
--- a/contrib/libmodplug/src/sndfile.cpp
+++ b/contrib/libmodplug/src/sndfile.cpp
@@ -161,6 +161,7 @@ BOOL CSoundFile::Create(LPCBYTE lpStream, DWORD dwMemLength)
 		 && (!ReadUlt(lpStream, dwMemLength))
 		 //&& (!ReadDMF(lpStream, dwMemLength))
 		 && (!ReadDSM(lpStream, dwMemLength))
+		 && (!ReadGDM(lpStream, dwMemLength))
 		 //&& (!ReadUMX(lpStream, dwMemLength))
 		 && (!ReadAMF(lpStream, dwMemLength))
 		 //&& (!ReadPSM(lpStream, dwMemLength))
